rqd/rqd/rqutil.py

```diff
@@ -145,13 +145,12 @@ def getHostIp():
 
 def getHostname():
     """Returns the machine's fully qualified domain name"""
-    if platform.system() == "Linux":
+    try:
         if rqd.rqconstants.RQD_USE_IP_AS_HOSTNAME:
             return getHostIp()
         else:
-            # This may not work in windows/mac, need to test
             return socket.gethostbyaddr(socket.gethostname())[0].split('.')[0]
-    else:
+    except socket.herror:
         return socket.gethostname()
```

Comment:
Could we also log a warning here to let the user know this is a fallback behavior? Otherwise it could be confusing as to why the `RQD_USE_IP_AS_HOSTNAME` setting isn't being used.